core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -597,21 +624,42 @@ class AuthorizerIntegrationTest extends KafkaServerTestHarness {
                              part: Int = part) {
     val records = new ArrayList[ConsumerRecord[Array[Byte], Array[Byte]]]()
     val maxIters = numRecords * 50
-    var iters = 0
-    while (records.size < numRecords) {
-      for (record <- consumer.poll(50).asScala) {
-        records.add(record)
+    
+    var caughtExc : Exception = null
+
+    val t = new java.lang.Thread() {
+      override def run () {
+        println("in consumeRecords.Thread.run")
+        try {
+          var iters = 0
+            while (records.size < numRecords) {
+                for (record <- consumer.poll(50).asScala) {
+                    records.add(record)
+                }
+                if (iters > maxIters)
+                    throw new TestTimeoutException("Failed to consume the expected records after " + iters + " iterations.")
+                    iters += 1
+            }
+        } catch {
+          case e : Exception => caughtExc = e
+        }
       }
-      if (iters > maxIters)
-        throw new IllegalStateException("Failed to consume the expected records after " + iters + " iterations.")
-      iters += 1
     }
+    t.start()
+    t.join(10000L)
+    
+    if (caughtExc != null)
+      throw caughtExc
+    
+    if (records.isEmpty()) 
+        throw new TestTimeoutException("Failed to consume the expected records after 10000 millis.")
```

Comment:
It sounds like we should have a helper test method for this functionality as there is similar code in another test (maybe in `TestUtils`). And as I said elsewhere, please use scala Futures.